### ISSUE TITLE
Pawn started 10 jobs in 10 ticks when pawn has assigned loadout and attempts animal handling with berries.

### ISSUE THREAD

**Issue by ProfoundDarkness** [skyarkhangel/CombatRealism#40]
The log text specifically:
"<Pawn> started 10 jobs in 10 ticks. lastJobGiver=, curJob.def=Wait, curDriver=Verse.AI.JobDriver_Wait"

This occurs with a new game with ONLY Core and Combat Realism loaded.  Combat Realism A16a.

This is remarkably similar to issue #5 though it only seems to happen when the pawn selects berries to handle the animal.  Corn, Potatoes, Meat, and Kibble DO NOT cause this to happen and everything proceeds smoothly.

When the pawn which elects to do animal handling on their own has a non-empty loadout assigned and chooses berries to handle the animal with the pawn gets stuck with the error message above when they try picking up the berries.

If said pawn is forced into the animal interaction they will pickup/drop the berries within a single tick and then move onto something else in the next tick.  (Using the numeric plus key to step forward tick by tick can hear the sound of an item drop and nothing appears in inventory).

Replication procedure:
- Start Rimworld with the following mods: Core, Combat Realism.
- Start a new game.  During game setup keep re-rolling a colonist until one with good Animal skill.
- Pause the game on start after colonists land (not strictly necessary).
- In game, spawn a stack of berries (stack 75).
- Ensure the best Animal handler pawn is the only one with the animal work type.
- Assign the animal handler pawn a new loadout, with say a knife.
- Pick an animal to tame/assign pet for training, make sure the pawn can tame/train it.
- Draft - undraft our pawn and then if game is paused, unpause.
- Observe the pawn decide to tame the animal and grab some berries.
- Observe the red error log about how the pawn started 10 jobs in 10 ticks when on the berries.

I haven't tested beyond a small variety of meat, potatoes, corn, and kibble just yet.  (and berries, obviously.)

**Comment by duncanburke** [comment]
Okay, I've reproduced this. Thanks for the detailed instructions.

The pawn is stuck on the berries and not completing any jobs.

I'll see what I can do.

**Comment by ProfoundDarkness** [comment]
So I stumbled on something interesting.  I modifiedberries to have a preferability of DesperateOnly, with the intention to not have pawns use berries in animal work.  Unfortunately the pawns still (rarely) use berries in animal interactions but fortunately no errors occur as the berries are collected and used in taming/training.  I've verified this by stripping all mods back except Core, CR, and my Berries mod (branched from CR).  Ran through the replication procedure and got no issues.

So seems like it has less to do with the berries directly and more to do with an interaction in CR and preferability.

**Comment by ProfoundDarkness** [comment]
Fascinating "new" tidbit (Without my mod to berries, have verified with just Core and CR)... 
If the following are true, on top of the replication steps in first post, then the behavior is IDENTICAL to issue #5 :
-Stockpile zone allowing berries.
-Work in numeric (priority) mode.
-Haul is higher priority than handle.

**Comment by ProfoundDarkness** [comment]
I didn't know of any other items with preferability=RawTasty, now I do: Milk.  Same behavior outlined so both Berries and Milk.  The temp workaround (mod) for berries also works for milk.

**Comment by Thorbane** [comment]
I think the problem is that the Update Loadout jobgiver and whatever jobgivers tell a pawn to pick up meals, drugs, or food items for taming are fighting.  They tell a pawn to put an item in their inventory, then the loadout jobgiver tells them to drop items that aren't in their loadout.

**Comment by carpediembr** [comment]
I think this issue got a grew a bit on this last update and it affect wardens as well. 

My warden have a loadout to carry 1 simple meal, and when trying to offer (throw the food on the floor) to a prisoner, he will pick it back up and go into a chain reaction.

**Comment by Thorbane** [comment]
Huh.  I thought prisoner meal were forbidden when dropped before.  That would be a fix this issue in wardens.

**Comment by SheakZarnn** [comment]
I'm not sure if it works that way, but may I suggest, that UpdateLoadout jobgiver should simply ignore items with positive nutrition? It should really solve many of the problems described. If it could potentially lead to inventories overflowing with food, then maybe ignoring two most nutritious items (one meal and another is food for taming/training)?

**Comment by ProfoundDarkness** [comment]
I just noticed in the commits that something was attempted to rectify this issue.  I tested Core and Combat Realism 1.6.9.6 with berries and milk using replication steps in my original post as well as with haul work priority being more important than handle priority.  Everything seemed to work fine :) .

The prisoner issue seems to be different.  My issue was very specific to animal handling and loadouts.  I'll put a new issue up with replication steps for that, closing as my particular issue seems solved.
